src/MassTransit/DependencyInjection/DependencyInjection/SetScopedConsumeContext.cs

```diff
@@ -0,0 +1,21 @@
+namespace MassTransit.DependencyInjection
+{
+    using System;
+
+
+    public class SetScopedConsumeContext :
+        ISetScopedConsumeContext
+    {
+        readonly Func<IServiceProvider, IScopedConsumeContextProvider> _factory;
+
+        public SetScopedConsumeContext(Func<IServiceProvider, IScopedConsumeContextProvider> factory)
```

Comment:
Is it a factory (it isn't creating anything)? Seems like more of a `providerProvider` :)